File: gcc/testsuite/gfortran.dg/predict-2.f90
! { dg-do compile }
! { dg-options "-O2 -fdump-tree-profile_estimate" }

subroutine test(block, array)
integer :: i,j, block(9), array(2)

do i = array(1), array(2), 2
    do j = array(1), array(2), 3
       block(i) = j
    end do
end do
end subroutine test

! { dg-final { scan-tree-dump-times "Fortran loop preheader heuristics of edge" 2 "profile_estimate" } }
! { dg-final { scan-tree-dump-times "loop guard" 0 "profile_estimate" } }

File: gcc/testsuite/gcc.dg/tree-ssa/predict-2.c
/* { dg-do compile } */
/* { dg-options "-O2 -fdump-tree-profile_estimate" } */
int test2();
int
test (int a, int b)
{
	if (__builtin_expect_with_probability (a, 0, 0.8)+__builtin_expect_with_probability (b, 5, 0.9) == 5)
		test2();
}
/* Combining two predictions together can not be done precisely, so check that result is DS theory.  */
/* { dg-final { scan-tree-dump-times "combined value predictions heuristics of edge .->.: 72.00" 1 "profile_estimate"} } */
